### PR TITLE
upgrade used cluster network addons operator version to v0.3.0

### PR DESCRIPTION
**Pull request by phoracek**
Introducing KubeMacPool support. SR-IOV, although available in v0.3.0,
is disabled, since it wasn't thoroughly tested.

Release: https://github.com/kubevirt/cluster-network-addons-operator/releases/tag/v0.3.0

### CODE REVIEW

**Comment by phoracek** [comment]
@rthallisey @djzager could you please take a look?

**Comment by lukas-bednar** [comment]
I believe you need to change also following file:
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/26cf89c470f26361e82ac0e4c2ff408339927db4/hack/defaults#L9

**Comment by phoracek** [comment]
@lukas-bednar thanks, should be fixed. Tested via `operator-test.sh`

**Comment by lukas-bednar** [comment]
@phoracek thanks looks good, looking forward to test it with different registry ;-)

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -259,7 +259,8 @@ func newNetworkAddonsForCR(cr *hcov1alpha1.HyperConverged) *networkaddons.Networ
 		Spec: networkaddons.NetworkAddonsConfigSpec{
 			Multus:          &networkaddons.Multus{},
 			LinuxBridge:     &networkaddons.LinuxBridge{},
-			ImagePullPolicy: NetworkAddonsImagePullPolicy,
+			KubeMacPool:     &networkaddons.KubeMacPool{},
+			ImagePullPolicy: v1.PullPolicy(NetworkAddonsImagePullPolicy),
```
Comment (myakove) [inline pkg/controller/hyperconverged/hyperconverged_controller.go]: What about sr-iov?

Comment (phoracek) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: Not fully integrated in 0.3.0. Will be added with 0.4.0.

Comment (myakove) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: I know, but I know @dankenigsberg want's it in anyway.

Comment (phoracek) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: **Edit** it won't be added in 0.4.0. We want to test it, however, it won't be enabled in HCO. If you want to try it, you have to edit NetworkAddonsConfig.Spec.Sriov directly.

Comment (myakove) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: OK

**Pull request merged**